### DISCUSSION TITLE
Support for date-fns 3.0.0

### DISCUSSION THREAD

**Discussion by JoepKockelkorn** [gpbl/react-day-picker#1997] [Support] [answered]
Very recently [date-fns 3.0.0 came out](https://blog.date-fns.org/v3-is-out/). I tried updating, but I got a peer dependency warning for react-day-picker which expects version `^2.28.0`. I created this issue just to get some attention for it.

**Comment by adaptive-shield-matrix** [comment]
I'm interested to see the v3 update of date-fns as well, because it promises substantial bundle size reduction.

Could you post the error?

By my reasoning updating the peerDependencies versions in package.json should be enough
```json
"peerDependencies": {
    "date-fns": "^2 || ^3",
    "react": "^16.8.0 || ^17.0.0 || ^18.0.0"
  },
```

**Reply by JoepKockelkorn** [reply]
I didn't try it at runtime, because I stopped the upgrade when I saw the warning. Maybe I'll try in the new year, happy holidays!

**Reply by gpbl** [reply]
Hey thanks for the heads up. We are investigating some breaking changes that may break the upgrade before pushing the peer dependencies. Thanks for your patience!

**Reply by adaptive-shield-matrix** [reply]
Have you seen/experienced any breaking changes due to date-fns 3.0 itself? 
The API seems pretty much unchanged. So it seems to be a pretty easy/fast fix.

**Reply by gpbl** [reply]
@adaptive-shield-matrix there was a minor difference for the type of the `firstWeekContainsDate` option used by some formatters in date-fns. I updated it in #2004. Beside this issue, the upgrade was easy.

**Comment by gpbl** [comment, marked as answer]
We added support for date-fns@3 in [version 8.10](https://github.com/gpbl/react-day-picker/releases/tag/v8.10.0). Please upgrade `react-day-picker` before upgrading `date-fns`. 

Thanks for your patience!

**Comment by adaptive-shield-matrix** [comment]
This reduced the bundle size from 113 kB to 84.2 kB

8.10.0
https://bundlejs.com/?q=react-day-picker%408.10.0
Bundle size is 84.2 kB -> 23.2 kB (gzip)

8.9.1
https://bundlejs.com/?q=react-day-picker%408.9.1
Bundle size is 113 kB -> 28.9 kB (gzip)

**Reply by gpbl** [reply]
...I'd be surprised to be related to date-fns (or possibly I don't understand how these bundle sizes are calculated). We upgraded other build dependencies before the release, so it could be because of those updates?

Thanks for looking at it!

💭I'd love to have a GitHub action adding the size report to a PR...

**Comment by adaptive-shield-matrix** [comment]
I really hoped for a smaller bundle size, especially since date-fns advertised being as small as 200 bytes.

"• Minimal size is now 200 bytes!"
https://twitter.com/kossnocorp/status/1736711759183094172

official release post
https://blog.date-fns.org/v3-is-out/

But it seems like `react-day-picker` uses a pretty wide range of apis so the bundle size reduction is not as great (as I have hoped).

**Reply by gpbl** [reply]
@adaptive-shield-matrix react-day-picker doesn't include in its output any code from date-fns, since it relies to a peer dependency. You should see for sure a difference with upgraded date-fns in an app that bundles date-fns@3. 

I was interested by your finding, why a smaller size? With a diff I can confirm the upgraded typescript/rollup dependencies produced a better build. Let's hope without any issue :)
